src/full/Agda/TypeChecking/Rules/LHS.hs

```diff
@@ -2121,8 +2096,7 @@ checkSortOfSplitVar dr a tel mtarget = do
       Right True  -> return ()
 
     splitOnPropError dr = softTypeError =<< do
-      liftTCM $ GenericDocError <$>
-        ("Cannot split on" <+> kindOfData dr <+> "in Prop unless target is in Prop")
+      liftTCM $ SplitInProp <$> kindOfData dr
```

Comment:
I'd prefer if the new `SplitInProp` error would not carry a `Doc` but a `DataOrRecord` and the `kindOfData` printing function was moved to the printing of the `SplitInProp` error.

Reply:
Sorry for late answer, I finally got back to this PR. I changed `AmbiguousConstructor` and `AmbiguousProjection` according to your suggestions, but I am having a small problem with `SplitInProp`. I am unable to make `DataOrRecord` a param for this error because I cannot use `DataOrRecord` in `Base.hs` because of circular dependency. Is it okay if I create a new data type `DataOrRecord` in `Base.hs` and just use that in the context of errors?

Reply:
Ah, indeed this is a problem, complicated that we have two definitions of `DataOrRecord`:
1. In `Syntax.Internal`.
2. In `TypeChecking.Rules.LHS`.

I suggest this refactoring:
- Factor out a parametric `DataOrRecord'` in `Syntax.Internal` so that the new definition there is `type DataOrRecord = DataOrRecord' PatternOrCopattern`.
- Define in `TC.Monad.Base` an instantiation of `DataOrRecord'` to be used instead of the `DataOrRecord` from `TC.Rules.LHS`.

This should allow you to refactor `SplitInProp` without getting into a cyclic dependency.
Does this make sense?

Reply:
Thank you for the suggestion!

After implementing this, the one thing that's still not clear is how do we use the constructor of `SplitInProp` in `TC.Rules.LHS`, when `DataOrRecord'` is not in scope because of hiding.

One solution I can think of is creating a function in `TC.Monad.Base` that somehow maps from `DataOrRecord` from `TC.Rules.LHS` to the respective instantiation of `DataOrRecord'`. Since LHS's `DataOrRecord` is not visible in `TC.Monad.Base`, this function would have to take the `(Maybe Induction)` param to do the mapping.

Alternatively we could make `DataOrRecord` from `TC.Rules.LHS` also an instantiation of `DataOrRecord'` but I am not sure if this would work.

Reply:
> Alternatively we could make DataOrRecord from TC.Rules.LHS also an instantiation of DataOrRecord' but I am not sure if this would work.

That is what I intended.  So there would be no need to hide the other `DataOrRecord` type.

Reply:
Thank you! I just pushed the changes. Can you please check?

Reply:
Thanks, all good!